lib/widgets/text.dart

```diff
@@ -0,0 +1,120 @@
+import 'dart:io';
+import 'dart:ui';
+import 'package:flutter/widgets.dart';
+
+/// A mergeable [TextStyle] with 'Source Code Pro' and platform-aware fallbacks.
+///
+/// Callers should also call [weightVariableTextStyle] and merge that in too,
+/// because for this font, we use "variable font" assets with a "wght" axis.
+///
+/// Example:
+///
+/// ```dart
+/// kMonospaceTextStyle
+///   .merge(const TextStyle(color: Colors.red))
+///   .merge(weightVariableTextStyle(context))
+/// ```
+final TextStyle kMonospaceTextStyle = TextStyle(
+  fontFamily: 'Source Code Pro',
+
+  // Oddly, iOS doesn't handle 'monospace':
+  //   https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/flutter.3A.20monospace.20font.20fallback/near/1570622
+  fontFamilyFallback: Platform.isIOS ? ['Menlo', 'Courier'] : ['monospace'],
+
+  inherit: true,
+);
+
+/// A mergeable [TextStyle] to use when the preferred font has a "wght" axis.
+///
+/// Some variable fonts can be controlled on a "wght" axis.
+/// Use this to set a value on that axis. It uses [TextStyle.fontVariations],
+/// along with a [TextStyle.fontWeight] that approximates the given "wght"
+/// for the sake of glyphs that need to be rendered by a fallback font
+/// (which might not offer a "wght" axis).
+///
+/// Use this even to specify normal-weight text, by omitting `wght`; then,
+/// [FontWeight.normal.value] will be used. No other layer applies a default,
+/// so if you don't use this, you may e.g. get the font's lightest weight.
+///
+/// Pass [context] to respect a platform request to draw bold text for
+/// accessibility (see [MediaQueryData.boldText]). This handles that request by
+/// using [wghtPlatformRequestsBold] or if that's null, [FontWeight.bold.value].
+///
+/// Example:
+///
+/// ```dart
+/// someTextStyle.merge(weightVariableTextStyle(context, wght: 250)
+/// ```
+///
+/// See also [FontVariation] for more background on variable fonts.
+// TODO(a11y) make `context` required when callers can adapt?
+TextStyle weightVariableTextStyle(BuildContext? context, {
+  int? wght,
+  int? wghtPlatformRequestsBold,
+}) {
+  int value = wght ?? FontWeight.normal.value;
+  if (context != null && MediaQuery.of(context).boldText) {
+    // The framework has a condition on [MediaQueryData.boldText]
+    // in the [Text] widget, but that only affects `fontWeight`.
+    // [Text] doesn't know where to land on the chosen font's "wght" axis if any,
+    // and indeed it doesn't seem updated to be aware of variable fonts at all.
+    value = wghtPlatformRequestsBold ?? FontWeight.bold.value;
+  }
+  assert(value >= 1 && value <= 1000); // https://fonts.google.com/variablefonts#axis-definitions
+
+  return TextStyle(
+    fontVariations: [FontVariation('wght', value.toDouble())],
```

Comment:
This seems like a sign the parameters should just have type `double` in the first place.

Reply:
From that page linked on the `1..1000` assert above—https://fonts.google.com/variablefonts#axis-definitions —it looks like the values on some axes are expected to "step" by less than one, making `double` necessary for those, but for `wght`, values are expected to step by one. That was my thinking behind using `int`, but `double` wouldn't be wrong, I suppose; it'd just let us give non-integer values that would get rounded, or floored, etc.

Reply:
I see, yeah.

Given that the underlying API takes `double`, though, I think it makes sense to have this take `double`, in order to remain as thin a layer as it can while doing its job.